src/mongo/mongoConnectionStrings.ts

```diff
@@ -3,20 +3,34 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+// Connection strings follow the following format (https://docs.mongodb.com/manual/reference/connection-string/):
+//   mongodb[+srv]://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
+// Some example connection strings:
+//   mongodb://dbuser:dbpassword@dbname.mlab.com:14118
+//   mongodb+srv://db1.example.net:27017,db2.example.net:2500/?replicaSet=test
+//   mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/database?ssh=true
+// Regex splits into three parts:
+//   Full match
+//   mongodb[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+//   [database]
+const parsePrefix = '^(mongodb(?:[+]srv)?:\/\/[^\/]*)';
+const parseDatabaseName = '\/?([^/?]+)?';
+const parseConnectionString = new RegExp(parsePrefix + parseDatabaseName);
```

Comment:
> Would be much easier to read if you break up the regex into multiple string constants

Not a blocker for the PR, but I disagree with @StephenWeatherford on this one. I think splitting it up makes it more prone to mistakes (plus "easier to read" is debatable). When you're typing out a RegExp in the form `/fdsafsdfasd/` you will get errors if you have delimiters in the wrong place, etc. but you don't get that with strings. For example only the bottom one here has a red squiggly:
![screen shot 2018-10-12 at 11 33 53 am](https://user-images.githubusercontent.com/11282622/46887547-c682e380-ce12-11e8-8938-c609b742856a.png)

Reply:
You'll get the error when you create the RegExp.

Reply:
And it's easier to read because:
1) you can easily see where the groups are
2) it becomes self-documentating

The longer it becomes, the more this is true.

Reply:
3. You can re-use pieces in similar regexes if necessary

Reply:
Given the shorter length, less important.  Your call.